HubNotificationHandler/src/test/java/com/eagle/hubnotifier/NotifyHandlerServiceImplTest.java

```diff
@@ -50,47 +41,83 @@
         MockitoAnnotations.initMocks(this);
     }
 
+    @Test
+    void handleNotifyKafkaTopicRequest() {
+        NotifyHookServiceImpl notifyHookService = mock(NotifyHookServiceImpl.class);
+        doNothing().when(notifyHookService).handleNotifyKafkaTopicRequest(Matchers.any(Map.class));
+    }
+
     @Test
     void handleTopicCreate() throws Exception {
-        Map<String, Object> topicData = new HashMap<>();
         HubUser user = mock(HubUser.class);
-        topicData.put(Constants.PARAM_TOPIC_TITLE_CONSTANT, Arrays.asList("New topic check"));
-        topicData.put(Constants.PARAM_TOPIC_TID_CONSTANT, Arrays.asList("1"));
-        topicData.put(Constants.PARAM_TOPIC_UID_CONSTANT, Arrays.asList("1"));
-        when(userRepository.findByKey("user")).thenReturn(user);
-        notifyHandlerService.handleTopicCreate(topicData);
+        NotifyHookServiceImpl notifyHookService = mock(NotifyHookServiceImpl.class);
+        when(configuration.getDiscussionCreateUrl()).thenReturn(DISCUSSION_BASE_URL);
+        when(userRepository.findByKey(USER_CONST)).thenReturn(user);
+        doNothing().when(notifyHookService).handleTopicCreate(Matchers.any(Map.class));
     }
 
     @Test
-    void sendNotification() {
+    void handleTopicReplyEvent() throws Exception {
+        HubUser user = mock(HubUser.class);
+        HubTopic topic = mock(HubTopic.class);
+        Map<String, List<String>> recipients = new HashMap<>();
+        Map<String, Object> values = new HashMap<>();
+        NotifyHookServiceImpl notifyHookService = mock(NotifyHookServiceImpl.class);
         NotificationEvent notificationEvent = mock(NotificationEvent.class);
-        doNothing().when(notifyHandlerServiceImpl).sendNotification(notificationEvent);
+        when(notificationEvent.getEventId()).thenReturn(EVENT_ID);
+        when(notificationEvent.getRecipients()).thenReturn(recipients);
+        when(notificationEvent.getRootOrg()).thenReturn(ROOT_ORG);
+        when(notificationEvent.getTagValues()).thenReturn(values);
+        when(notificationEvent.getTargetData()).thenReturn(values);
+        when(configuration.getDiscussionCreateUrl()).thenReturn(DISCUSSION_BASE_URL);
+        when(hubTopicRepository.findByKey(TOPIC_DATA)).thenReturn(topic);
+        when(userRepository.findByKey(USER_CONST)).thenReturn(user);
+        doNothing().when(notifyHookService).handleTopicReplyEvent(Matchers.any(Map.class));
+    }
+
+    @Test
+    void handleTopicUpvoteEvent() {
+        HubPost post = mock(HubPost.class);
+        HubUser user = mock(HubUser.class);
+        NotifyHookServiceImpl notifyHookService = mock(NotifyHookServiceImpl.class);
+        when(hubPostRepository.findByKey(POST_DATA)).thenReturn(post);
+        List<HubUser> userList = new ArrayList<>();
+        userList.add(user);
+        when(userRepository.findByUUIDS(Arrays.asList("1", "2"))).thenReturn(userList);
+        doNothing().when(notifyHookService).handleTopicUpvoteEvent(Matchers.any(Map.class));
```

Comment:
Please look into this again. using this code it will not do any operation when notifyHookService class will be called.